### ISSUE TITLE
Parameter setting in BatchReNorm

### ISSUE THREAD

**Issue by John1231983** [seokhoonboo/caffe-boo#3]
Thanks for sharing your work!

I am training my network using your BatchReNorm implementation. I have two question for the layers:

1. Did you perform test the correction of implementation in (test_batch_renorm_layer.cpp) -gradient checking? I just replace BatchNorm cheking by BatchReNorm and it passed. However, if I perform it in the real network (like resnet), the GPU memory increase 1GB after 1000 iterations. For example (980 iteration, memory is 5GB, but in 1000 iterations it is 6.1GB). I guess you did not free memory yet. Please check it

2. For parameter setting, I will use default `r` and `d`. Do we need set up any parameters? In the deploy.prototxt, we will use `use_global_stats: true`, RIGHT?

```
layer {
  name: "brenorm1"
  type: "BatchReNorm"
  bottom: "conv1"
  top: "brenorm1"
  param {
    lr_mult: 0
    decay_mult: 0
  }
  param {
    lr_mult: 0
    decay_mult: 0
  }
  param {
    lr_mult: 0
    decay_mult: 0
  }
  batch_renorm_param {
    use_global_stats: false
  }
}
```

3. I used above setting and loss some time grow up rapidly. I think your implementation has some issue. Please check it

**Comment by strawberryfg** [comment]
Hello @John1231983 can you kindly provide any screenshot about training loss (curve)? And will the loss continue to decrease after growing up? How much does it affect validation/testing performance? Thanks! Much appreciated!!!
